k4Reco/DDPlanarDigi/components/DDPlanarDigi.h

```diff
@@ -0,0 +1,138 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef DDPLANARDIGI_H
+#define DDPLANARDIGI_H
+
+#include "Gaudi/Accumulators/Histogram.h"
+#include "Gaudi/Property.h"
+#include "GaudiAlg/Transformer.h"
+#include "GaudiKernel/RndmGenerators.h"
+#include "k4FWCore/BaseClass.h"
+#include "k4Interface/IGeoSvc.h"
+#include "k4Interface/IUniqueIDGenSvc.h"
+
+#include "edm4hep/EventHeaderCollection.h"
+#include "edm4hep/MCRecoTrackerHitPlaneAssociationCollection.h"
+#include "edm4hep/SimTrackerHitCollection.h"
+#include "edm4hep/TrackerHitPlaneCollection.h"
+
+#include "DDRec/SurfaceManager.h"
+
+#include <random>
+#include <string>
+#include <vector>
+
+/** ======= DDPlanarDigi ========== <br>
+ * Creates TrackerHits from SimTrackerHits, smearing them according to the input parameters. 
+ * The positions of "digitized" TrackerHits are obtained by gaussian smearing positions
+ * of SimTrackerHits perpendicular and along the ladder according to the specified point resolutions. 
+ * The geometry of the surface is retreived from DDRec::Surface associated to the hit via cellID.
+ * 
+ * 
+ * <h4>Input collections and prerequisites</h4> 
+ * Processor requires a collection of SimTrackerHits <br>
+ * <h4>Output</h4>
+ * Processor produces collection of smeared TrackerHits<br>
+ * @param SimTrackHitCollectionName The name of input collection of SimTrackerHits <br>
+ * (default name VXDCollection) <br>
+ * @param TrackerHitCollectionName The name of output collection of smeared TrackerHits <br>
+ * (default name VTXTrackerHits) <br>
+ * @param ResolutionU resolution in direction of u (in mm) <br>
+ * (default value 0.004) <br>
+ * @param ResolutionV Resolution in direction of v (in mm) <br>
+ * (default value 0.004) <br>
+ * @param IsStrip whether the hits are 1 dimensional strip measurements <br>
+ * (default value false)
+ * @param Ladder_Number_encoded_in_cellID ladder number has been encoded in the cellID <br>
+ * (default value false) <br>
+ * @param Sub_Detector_ID ID of Sub-Detector using UTIL/ILDConf.h from lcio <br>
+ * (default value lcio::ILDDetID::VXD) <br>
+ * <br>
+ * 
+ * @author F.Gaede CERN/DESY, S. Aplin DESY
+ * @date Dec 2014
+ * Originally in https://github.com/iLCSoft/MarlinTrkProcessors/blob/master/source/Digitisers/include/DDPlanarDigi.h
+ */
+
+using SimTrackerHitCollection = edm4hep::SimTrackerHitCollection;
+using Header                  = edm4hep::EventHeaderCollection;
+
+using TrackerHitPlaneColl = edm4hep::TrackerHitPlaneCollection;
+using Association         = edm4hep::MCRecoTrackerHitPlaneAssociationCollection;
```

Comment:
These are currently in the global namespace and some of them are very generic and could potentially collide. Since these are the first Functional algorithms that we create, and people will use them as templates, I think we should take care of avoiding some of these mistakes. Somewhat related: Can Algorithms live in namespaces without breaking Gaudi?

Reply:
Yes, maybe I got used to this but I removed them since they are useless.